js/components/utils.js

```diff
@@ -431,34 +431,38 @@ var Utils = {
    * @function arrayDeepMerge
    * @param {Array} target - An array that will receive new items if additional items are passed
    * @param {Array} source - An array containing additional items to merge into target
-   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge function
+   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge, swap, unionBy, clone
    * @returns {Array} new merged array with items from both target and source
    */
   arrayDeepMerge: function(target, source, optionsArgument) {
+    var targetArray = optionsArgument.swap ? source : target;
+    var sourceArray = optionsArgument.swap ? target : source;
     var self = this;
     var destination = [];
-    destination = source.slice();
-    target.forEach(function(targetItem, i) {
+    destination = sourceArray.slice();
```

Comment:
the [deep merge lib](https://github.com/KyleAMathews/deepmerge/blob/master/index.js#L64) we are using does that.

this custom `arrayDeepMerge` function is passed as a callback to deepmerge function and only used if there is an array in the object that needs to be merged.

Also see - https://github.com/KyleAMathews/deepmerge#arraymerge